### ISSUE TITLE
What's the idea behind default SeldonDeployment readinessProbe for gRPC service

### ISSUE THREAD

**Issue by zyxue** [SeldonIO/seldon-core#3941]
I get a feeling some container in my `SeldonDeployment` is not ready when upstream app is sending requests to it, seeing error like 

```
grpc._channel._InactiveRpcError: <_InactiveRpcError of RPC that terminated with:
	status = StatusCode.UNAVAILABLE
	details = "upstream connect error or disconnect/reset before headers. reset reason: connection failure"
	debug_error_string = "{"created":"@1645125206.023425744","description":"Error received from peer ipv4:10.48.13.4:5001","file":"external/com_github_grpc_grpc/src/core/lib/surface/call.cc","file_line":1063,"grpc_message":"upstream connect error or disconnect/reset before headers. reset reason: connection failure","grpc_status":14}"
```

but just for a few requests at the beginning.

And I see the default `readinessProbe` for each container in `SeldonDeployment` is like
```
    readinessProbe:
      failureThreshold: 3
      initialDelaySeconds: 20
      periodSeconds: 5
      successThreshold: 1
      tcpSocket:
        port: 9500
      timeoutSeconds: 1
```

I wonder why it's set this way, why use tcpSocket at port 9500 in particular? Why not use https://github.com/grpc-ecosystem/grpc-health-probe, which appears to be more commonly used in probing readiness of gRPC service?

**Comment by cliveseldon** [comment]
Might be partially solved by #3945

**Comment by zyxue** [comment]
```
 420   │     readinessProbe:
 421   │       failureThreshold: 3
 422   │       httpGet:
 423   │         path: /app-health/combiner/readyz
 424   │         port: 15020
 425   │         scheme: HTTP
 426   │       initialDelaySeconds: 20
 427   │       periodSeconds: 5
 428   │       successThreshold: 1
 429   │       timeoutSeconds: 1
```
wondering where is the path ` /app-health/combiner/readyz` set?
